### PR TITLE
Focus View (GitHub only)

### PR DESCRIPTION
**Pull request by jdgarcia**
The signed out and user row (when signed in) UIs have been redone

Focus View only shows up if you have GitHub connected, otherwise it's just a blank screen (for now, next PR will be "connect an integration" UI).

### CODE REVIEW

**Review by gitkrakel — CHANGES_REQUESTED** [review]
### Layout on Firefox ESR (115.x) is completely broken

<details>
<summary>Sign-in page</summary>

Screenshot from 2024-04-15 10-25-01
</details>

<details>
<summary>Focus View</summary>

Screenshot from 2024-04-15 10-31-54
</details>

### (optional) Nits with layout when permissions are requested

<details>
<summary>Scrollbar in Focus View when permissions are requested</summary>

Screenshot from 2024-04-15 10-35-20
</details>

<details>
<summary>Unnecessary word wrap when cloud and self-hosted are both requested</summary>

Screenshot from 2024-04-15 10-36-40
</details>

**Comment by jdgarcia** [comment]
@gitkrakel well, the good news is that the layout is broken simply because Firefox ESR doesn't support nested CSS. The bad news is I can't use nested CSS which I was happy to be able to use 😢 

> Scrollbar in Focus View when permissions are requested

That's intentional. The popup has a max height enforced by the browser, and I opted to make the permission banner and the user bar static, and everything else scrollable.

> Unnecessary word wrap when cloud and self-hosted are both requested

That's a consequence of the font size and weight used in the Figma design.

File: src/popup/components/FocusView.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+	useEffect(() => {
+		const loadData = async () => {
+			const githubToken = await fetchProviderToken('github');
+			if (!githubToken) {
+				setLoadingPullRequests(false);
+				return;
+			}
+
+			const providerClient = new GitHub({ token: githubToken.accessToken });
```
[the remark this answers is not in the document]
Comment (jdgarcia) [reply src/popup/components/FocusView.tsx:48]: I'm gonna have to update this almost immediately anyway, I don't think it matters if it's in this PR or the next.

**Comment by gitkrakel** [comment]
@jdgarcia So I looked into CSS nesting more, and a couple notes:

1. The current Firefox ESR does in fact support nesting, but it's disabled by default. The user needs to go into `about:config` -> `layout.css.nesting.enabled` to toggle it on, and then the Focus View seems to work as intended:

image

2. CSS nesting is enabled by default starting in Firefox 117: https://www.mozilla.org/en-US/firefox/117.0/releasenotes/. Therefore, regular releases of Firefox already support it
3. The next Firefox ESR is 128.0, which comes out in July 2024 (https://whattrainisitnow.com/calendar/). So if CSS nesting is something you feel strongly about, I think it's fine to compromise and tell Firefox ESR 115 users to enable the `about:config` flag (they should be a small fraction of the userbase anyway)

**Pull request merged**